packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2030,101 +2030,109 @@ mixin WidgetInspectorService {
         'result': result,
       });
     }
+
+    Map<String, Object>? additionalCallback(DiagnosticsNode node, InspectorSerializationDelegate delegate) {
+      final Object? value = node.value;
+      final RenderObject? renderObject = value is Element ? value.renderObject : null;
+      if (renderObject == null) {
+        return null;
+      }
+      assert(delegate.addAdditionalPropertiesCallback == additionalCallback);
+      // The new delegate for collecting properties on `renderObject`.
+      // Setting `subtreeDepth` to 0 to skip child RenderObjects. Also removes
+      // the additional callback from `delegate` since we should only collect
+      // RenderObjects from the element tree.
+      final DiagnosticsSerializationDelegate newDelegate = InspectorSerializationDelegate(
```

Comment:
why can we no longer use delegate.copyWith?

Reply:
Not sure if it makes sense to add the callback as an argument of `copyWith`. Also I don't think we can use `copyWith` to set something to `null`, since `null` is used to represent "no changes" in `copyWith`.

Reply:
Can you add the addAdditionalPropertiesCallback parameter with an explicit null here (`addAdditionalPropertiesCallback: null`)? And also add a comment to describe why we are using the InspectorSerializationDelegate constructor instead of `copyWith`?

Reply:
The default value is null so there seems to be a lint preventing that (`avoid_redundant_argument_values`).

Reply:
You should be able to ignore the lint for that line:
```
// We are using the [InspectorSerializationDelegate] constructor instead of
// [InspectorSerializationDelegate.copyWith] so that the [addAdditionalPropertiesCallback]
// property can be set to null.
// ignore: avoid_redundant_argument_values
addAdditionalPropertiesCallback: null
```

If you can't ignore the lint, then just add the comment.

Reply:
this should use copyWith